ceph_deploy/hosts/debian/__init__.py

```diff
@@ -16,6 +16,8 @@ def choose_init(module):
 
     Returns the name of a init system (upstart, sysvinit ...).
     """
+    if module.conn.remote_module.path_exists("/lib/systemd/system/ceph.target"):
+        return 'systemd'
```

Comment:
yeah, the problematic cases are jessie with non-systemd and ubuntu 
15.04...


On Thu, 10 Sep 2015, Travis Rhoden wrote:

> 
> In ceph_deploy/hosts/debian/__init__.py:
> 
> > @@ -16,6 +16,8 @@ def choose_init(module):
> >  
> >      Returns the name of a init system (upstart, sysvinit ...).
> >      """
> > +    if module.conn.remote_module.path_exists("/lib/systemd/system/ceph.target"):
> > +        return 'systemd'
> 
> Since it is a runtime option, yes you are right. It's something I've talked about with @dachary before that I know
> ceph-deploy needs to switch over to ceph-detect-init.
> 
> That is the longer-term fix, for sure.
> 
> For now, the only time we would get in trouble is people running Infernalis RC on Debian and changing the init at runtime?
> Or is that even on older distros (14.04) that Infernalis packages are going to install ceph.target?
> 
> ?
> Reply to this email directly or view it on GitHub.[AAabh0O3GXLbgPO7qaF5DY1YwBW_tPuTks5owglJgaJpZM4F7ZGn.gif]
> 
> 
>